bundles/org.openhab.ui/web/src/assets/definitions/blockly/blocks-scripts.js

```diff
@@ -283,11 +283,11 @@ export default function defineOHBlocks_Scripts (f7, isGraalJs, scripts) {
     if (contextInfo === 'ruleUID') return ['ctx.ruleUID', javascriptGenerator.ORDER_ATOMIC]
     if (contextInfo === 'itemState' || contextInfo === 'oldItemState' || contextInfo === 'itemCommand') {
       if (type === 'asNumber') {
-        return [`parseFloat(event.${contextInfo}.toString())`, javascriptGenerator.ORDER_ATOMIC]
+        return [`parseFloat(event.${contextInfo}?.toString())`, javascriptGenerator.ORDER_ATOMIC]
       } else if (type === 'asQuantity') {
-        return [`Quantity(event.${contextInfo}.toString())`, javascriptGenerator.ORDER_ATOMIC]
+        return [`Quantity(event.${contextInfo}?.toString())`, javascriptGenerator.ORDER_ATOMIC]
```

Comment:
Same as above:

```suggestion
        return [`event.${contextInfo} !== undefined ? Quantity(event.${contextInfo}.toString() : undefined)`, javascriptGenerator.ORDER_ATOMIC]
```

Quantity will throw an error if it gets passed in undefined, to adding optional chaining inside would only move the source of the error.